### PR TITLE
fix issue #33 with recursive read

### PR DESCRIPTION
**Pull request by MrDocx**

### CODE REVIEW

File: src/main/java/io/github/anthorx/parquet/sql/api/SQLParquetReader.java [hunk 1]
```diff
@@ -100,19 +100,20 @@ public Record read() throws IOException {
...
   }
 
   private Record readFromNextParquetReader() throws IOException {
+    Record result = null;
     if (parquetReaderIterator.hasNext()) {
       currentParquetReader = parquetReaderIterator.next();
-      return currentParquetReader.read();
-    } else {
-      return null;
+      result = read();
```
Comment (anthoRx) [inline src/main/java/io/github/anthorx/parquet/sql/api/SQLParquetReader.java:113]: I think it's better for simplicity and comprehension if you move the recursive call inside the read method.

**Review by anthoRx — CHANGES_REQUESTED** [review]

File: pom.xml [hunk 2]
```diff
@@ -190,7 +190,7 @@
                         </goals>
                         <configuration>
                             <gpgArguments>
-                                <arg>--pinentry-mode</arg>
+                                <arg>&#45;&#45;pinentry-mode</arg>
```
[the remark this answers is not in the document]
Comment (MrDocx) [reply pom.xml]: I had to comment this plugin for local mvn install, so the IDE had to replace the `-` to comment.
It's now fixed

File: src/main/java/io/github/anthorx/parquet/sql/api/SQLParquetReader.java (hunk 1 above)
Comment (MrDocx) [reply src/main/java/io/github/anthorx/parquet/sql/api/SQLParquetReader.java]: It's now better (with only one method)

File: src/main/java/io/github/anthorx/parquet/sql/api/SQLParquetReader.java [hunk 3]
```diff
@@ -93,26 +93,18 @@ public MessageType getSchema() {
   }
 
   public Record read() throws IOException {
-    if (currentParquetReader == null) {
-      return readFromNextParquetReader();
-    }
-
-    Record result = currentParquetReader.read();
+    Record resultParquetReader = null;
```
Comment (anthoRx) [inline src/main/java/io/github/anthorx/parquet/sql/api/SQLParquetReader.java:96]: resultRecord or recordResult is more appropriate here as this variable is not a parquetReader object but a Record one.

Comment (MrDocx) [reply src/main/java/io/github/anthorx/parquet/sql/api/SQLParquetReader.java]: Right so I fixed it (and I squashed all commits and did a push force)

**Review by anthoRx — APPROVED** [review]

**Pull request merged**